File: 3n+1/3n+1.py
def get_cycle(num: int):

    count = 1
    while num != 1:
        if num % 2 == 1:
            num = num * 3 + 1
        else:
            num = num / 2
        count += 1
    return count

while True:
    try:
        a, b = input().split(' ')
        a = int(a)
        b = int(b)

        max_cycle = 0

        for i in range(min(a,b), max(a,b)+1):
            
            tmp_cycle = get_cycle(i)
            
            if tmp_cycle > max_cycle:
                max_cycle = tmp_cycle
        
        print(f"{a} {b} {max_cycle}")


    except:
        break

File: 3n+1/3n+1.c
#include <stdio.h>

#define max(a,b) ((a)>(b)?(a):(b))
#define min(a,b) ((a)<(b)?(a):(b))

int answer[1000000] = {0};

int get_cycle(int num)
{
	if(answer[num] != 0)
	{
		return answer[num];
	}

	int count = 1;
	while(num != 1)
	{
		if(num & 0x0001)
		{
			num = num * 3 + 1;
		}
		else
		{
			num >>= 1;
		}
		count++;
	}
	answer[num] = count;
	
	return count;
}

int main()
{
	int a, b;
	
	while(scanf("%d %d", &a, &b) == 2)
	{
		int max_cycle = 0;

		for(int num = min(a, b); num <= max(a, b); num++)
		{
			int tmp_cycle = get_cycle(num);
			if(tmp_cycle > max_cycle)
			{
				max_cycle = tmp_cycle;
			}
		}
		printf("%d %d %d\n", a, b, max_cycle);
	}
	
	return 0;
}